content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+**Everything else in the IPFS ecosystem builds on top of this core concept; linked, addressable content is the fundamental connecting element that makes the rest work.**
+
+### 2 \\ DAGs
+
+<img alt="IPFS Stack - Applications" src="../assets/ipfs_stack-apps.png" width="200px" />
+
+IPFS and many other distributed systems take advantage of a concept called [directed acyclic graphs](https://en.wikipedia.org/wiki/Directed_acyclic_graph), or DAGs. DAG structures are what allow distributed nodes to connect to each other in computationally reasonable ways, and are how you can build trees of content-addressed nodes.
+
+As mentioned above, IPFS has its own preferences and conventions about how data should be broken up into DAGs that optimize for its specific use cases. You can do this chunking and structuring in lots of different ways. For example, Git is a DAG that has many versions of your repo inside of it. Its DAG tree doesn't behave in the same way the IPFS DAG does.
+
+IPFS first separates your content into _blocks_. If you had a huge file with a single CID, you'd have to transfer the whole CID every time someone requested it. But Instead, you can break it into blocks, and transfer it block by block instead of one massive chunk at a time. Each block might come from a different source, even! This is a much more efficient way to transmit data. (If you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)
+
+Another great feature of the DAG and breaking content into blocks is that if you have two similar files, they can share the DAG; ie, different trees can reference the same data. For example, if you copyedit a document and change a few commas here and there, IPFS knows to only create new blocks for those small changes. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) much more efficient because you only need to transfer the parts that are new or have changed instead of creating entirely new files each time.
+
+DAGs are a bit of a ["turtles all the way down"](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Turtles_all_the_way_down.html) scenario; that is, **everything** has a CID! You've got a file that has a CID. What if there are several files in a folder? That folder has a CID, and that CID contains the CIDs of the files underneath. In turn, those files are made up of blocks, and each of those blocks has a CID. You can see how a file system on your computer could just be a DAG. You can also, hopefully, how DAG trees start to form.
```

Comment:
At some point we should have started talking about Merkle-DAGs. DAGs may not necessarily use content-addressing.

```suggestion
DAGs are a bit of a ["turtles all the way down"](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Turtles_all_the_way_down.html) scenario; that is, **everything** has a CID! You've got a file: that has a CID. What if there are several files in a folder? That folder has a CID, and that CID contains the CIDs of the files underneath. In turn, those files are made up of blocks, and each of those blocks has a CID. You can see how a filesystem on your computer could just be a DAG. You can also, hopefully, how DAG trees start to form.
```

Reply:
> That folder has a CID, and that CID contains the CIDs of the files underneath

No, the CID addresses a special file that has the CIDs of the files in that folder (and their actual filenames).

Reply:
>At some point we should have started talking about Merkle-DAGs. DAGs may not necessarily use content-addressing.

I wasn't sure if that level of detail was important for this intro. I'm going to take another pass, tho, now that I've seen your and Oli's edits.

>No, the CID addresses a special file that has the CIDs of the files in that folder (and their actual filenames).

I'm not sure that amount of detail is useful for this high-level explainer. For example, our [IPLD explorer](https://explore.ipld.io/#/explore/QmSnuWmxptJZdLJpKRarxBMS2Ju2oANVrgbr2xWbie9b2D) doesn't show the special file that is the *method* IPFS just shows CIDs linking to CIDs, not the internal method for making that happen. What do you think?

Reply:
I will re-read. But the CIDs do not contain CIDs, they point to DAG nodes that contain CIDs.